h2o-algos/src/main/java/hex/modelselection/ModelSelectionUtils.java

```diff
@@ -102,42 +104,138 @@ public static void setBitSet(BitSet predBitSet, int[] currIndices) {
         return crossProductMatrix;
     }
+    
+    public static double[] dropIgnoredCols(GLMTask.GLMIterationTask gtask, List<Integer> ignoredCols) {
+        Gram gram = gtask.getGram();
+        int[] droppedCols = ignoredCols.stream().mapToInt(x->x).toArray();
+        gram.dropCols(droppedCols);
+        return ArrayUtils.removeIds(gtask.getXY(), droppedCols);
+    }
+    
+    /***
+     * The duplicated columns generated by qr-cholesky is at the level of coefficients.  This means for enum predictors
+     * there could be multiple coefficients, one for each level of the enum level.  For enum predictors, I will 
+     * remove the predictor only if all its columns are duplicated.  This make it a little more complicated.
+     */
+    public static List<Integer> findFullDupPred(DataInfo dinfo, List<Integer> ignoredCols, List<String> ignoredPredInd,
+                                                String[] prednames) {
+        List<Integer> ignoredColsCopy = new ArrayList<>(ignoredCols);
+        List<Integer> fullIgnoredCols = new ArrayList<>();
+        int[] catOffsets = dinfo._catOffsets;
+
+        if (dinfo._cats > 0) {   // there are enum columns in dataset
+            int catOffsetsLen = catOffsets.length;
+            for (int index = 1; index < catOffsetsLen; index++) {
+                final int counter = index;
+                List<Integer> discarded = ignoredColsCopy.stream().filter(x -> x < catOffsets[counter]).collect(Collectors.toList());
+                if ((discarded != null) && (discarded.size() == (catOffsets[index]-catOffsets[index-1]))) {  // full enum predictors found in ignored columns
+                    fullIgnoredCols.addAll(discarded);
+                    ignoredPredInd.add(prednames[index-1]);
+                }
+                if (discarded != null && discarded.size() > 0)
+                    ignoredColsCopy.removeAll(discarded);
+            }
+        }
+        if (ignoredColsCopy != null && ignoredColsCopy.size()>0) {
+            int offsetNum = dinfo._numOffsets[0]-dinfo._cats;
+            ignoredPredInd.addAll(ignoredColsCopy.stream().map(x -> prednames[x-offsetNum]).collect(Collectors.toList()));
+            fullIgnoredCols.addAll(ignoredColsCopy);    // add all remaining numerical ignored predictors columns
+        }
+        return fullIgnoredCols;
+    }
+    
+    public static GLMTask.GLMIterationTask genGramCheckDup(Key jobKey, DataInfo dinfo, ArrayList<Integer> ignoredCols, 
+                                                    ModelSelectionModel.ModelSelectionParameters parms) {
+        double[] beta = new double[dinfo.coefNames().length];
+        beta = Arrays.stream(beta).map(x -> 1.0).toArray(); // set coefficient to all 1
+        GLMTask.GLMIterationTask gtask = new GLMTask.GLMIterationTask(jobKey, dinfo, new GLMModel.GLMWeightsFun(gaussian,
+                GLMModel.GLMParameters.Link.identity, 1, 0.1, 0.1), beta).doAll(dinfo._adaptedFrame);
```

Comment:
@tomasfryda Yes, this is only for Gaussian families.  The 1, 0.1 and 0.1 are specified for other families (I believe it is Tweedie (your favourite) and negative binomial?).  In theory, it can be any value because we are dealing with Gaussian here, they will be ignored.  The main purpose here is to generate the gram matrix which is transpose(W)*weights*W and Transpose(X)*Y.  Anything else will not be used and will be ignored.